omniduct/filesystems/s3.py

```diff
@@ -41,27 +41,34 @@ def _init(self, bucket=None, aws_profile='default', path_separator='/'):
         self.bucket = bucket
         self.aws_profile = aws_profile
         self.__path_separator = path_separator
+        self._client = None
 
-        self._setup_session()  # Ensure self.host is updated with correct AWS region
-
-    def _setup_session(self):
+        # Ensure self.host is updated with correct AWS region
         import boto3
-        s = boto3.Session(profile_name=self.aws_profile)
-        self._session = s
-        self._client = s.client('s3')
-        self._resource = s.resource('s3')
-        self.host = 'autoscaling.{}.amazonaws.com'.format(self._client.meta.region_name)
+        self.host = 'autoscaling.{}.amazonaws.com'.format(
+            boto3.Session(profile_name=self.aws_profile).region_name
+        )
 
     def _connect(self):
         from opinel.utils.credentials import read_creds
 
         # Refresh access token, and attach credentials to current object for debugging
         self._credentials = read_creds(self.aws_profile)
 
-        # Update AWS session, client and resource objects
-        self._setup_session()
+        import boto3
```

Comment:
perhaps we should not use opinel at all?

Reply:
Opinel provides support for MFA, whereas I don't think boto3 does. Is that true? It was necessary at the time to integrate with Airbnb's platform. I'm reluctant to drop it, since it provides a superset of the credential detection functionality.